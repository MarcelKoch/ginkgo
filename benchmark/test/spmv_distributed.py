#!/usr/bin/env python3
import test_framework
base_flags = ["spmv/distributed/spmv_distributed"]
# check that all input modes work:
# parameter
test_framework.compare_output_distributed(base_flags + ["-input", '[{"size": 100, "stencil": "7pt", "comm_pattern": "stencil"}]'],
                                          expected_stdout="spmv_distributed.simple.stdout",
                                          expected_stderr="spmv_distributed.simple.stderr",
                                          num_procs=3)

# stdin
test_framework.compare_output_distributed(base_flags,
                                          expected_stdout="spmv_distributed.simple.stdout",
                                          expected_stderr="spmv_distributed.simple.stderr",
                                          num_procs=3,
                                          stdin='[{"size": 100, "stencil": "7pt", "comm_pattern": "stencil"}]')

# input file
test_framework.compare_output_distributed(base_flags + ["-input", str(test_framework.sourcepath / "input.distributed_mtx.json")],
                                          expected_stdout="spmv_distributed.simple.stdout",
                                          expected_stderr="spmv_distributed.simple.stderr",
                                          num_procs=3)

# profiler annotations
test_framework.compare_output_distributed(base_flags + ["-input", '[{"size": 100, "stencil": "7pt", "comm_pattern": "stencil"}]', '-profile', '-profiler_hook', 'debug'],
                                          expected_stdout="spmv_distributed.profile.stdout",
                                          expected_stderr="spmv_distributed.profile.stderr",
                                          num_procs=3)
