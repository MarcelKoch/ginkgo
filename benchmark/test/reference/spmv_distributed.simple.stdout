
[
    {
        "size": 81,
        "stencil": "7pt",
        "comm_pattern": "stencil",
        "spmv": {
            "csr-csr": {
                "storage": 6420,
                "max_relative_norm2": 1.0,
                "time": 1.0,
                "repetitions": 10,
                "completed": true
            }
        },
        "nnz": 144,
        "optimal": {
            "spmv": "csr-csr"
        }
    }
]
